code/modules/cm_preds/yaut_weapons.dm

```diff
@@ -417,76 +421,145 @@
-				playsound(loc, 'sound/weapons/bladeslice.ogg', 25)
-				create_leftovers(victim, has_meat = FALSE, skin_amount = 3)
-				for(var/limb in victim.limbs)
-					victim.apply_damage(18, BRUTE, limb, sharp = FALSE)
-				victim.remove_overlay(UNDERWEAR_LAYER)
-				victim.f_style = "Shaved"
-				victim.update_hair() //then rip the beard off along the skin
-				victim.add_flay_overlay(stage = 2)
-
-				if(do_after(user, 4 SECONDS, INTERRUPT_ALL, BUSY_ICON_HOSTILE, victim))
-					to_chat(user, SPAN_WARNING("You completely flay [victim], sloppily ripping most remaining flesh and skin off the body. Use rope to hang them from the ceiling."))
-					playsound(loc, 'sound/weapons/wristblades_hit.ogg', 25)
-					create_leftovers(victim, has_meat = TRUE, skin_amount = 2)
-					for(var/limb in victim.limbs)
-						victim.apply_damage(22, BRUTE, limb, sharp = FALSE)
-					for(var/obj/item/item in victim)
-						victim.drop_inv_item_to_loc(item, victim.loc, FALSE, TRUE)
-
-					victim.status_flags |= PERMANENTLY_DEAD
-					victim.add_flay_overlay(stage = 3)
+
+		if(FLAY_STAGE_STRIP)
+			user.visible_message(SPAN_DANGER("<B>[user] jabs \his [tool.name] into [victim]'s cuts, prying, cutting, then tearing off large areas of skin. The remainder hangs loosely.</B>"),
+				SPAN_DANGER("<B>You jab your [tool.name] into [victim]'s cuts, prying, cutting, then tearing off large areas of skin. The remainder hangs loosely.</B>"))
+			playsound(user.loc, 'sound/weapons/bladeslice.ogg', 25)
+			create_leftovers(victim, has_meat = FALSE, skin_amount = 3)
+			flaying_stage = FLAY_STAGE_SKIN
+			for(var/limb in victim.limbs)
+				victim.apply_damage(18, BRUTE, limb, sharp = TRUE)
+			victim.remove_overlay(UNDERWEAR_LAYER)
+			victim.drop_inv_item_on_ground(victim.get_item_by_slot(WEAR_BODY)) //Drop uniform, belt etc as well.
+			victim.f_style = "Shaved"
+			victim.update_hair() //then rip the beard off along the skin
+			victim.add_flay_overlay(stage = 2)
+
+		if(FLAY_STAGE_SKIN)
+			user.visible_message(SPAN_DANGER("<B>[user] completely flays [victim], pulling the remaining skin off of \his body like a glove!</B>"),
+				SPAN_DANGER("<B>You completely flay [victim], pulling the remaining skin off of \his body like a glove.\nUse rope to hang \him from the ceiling.</B>"))
+			playsound(user.loc, 'sound/weapons/wristblades_hit.ogg', 25)
+			create_leftovers(victim, has_meat = TRUE, skin_amount = 2)
+			for(var/limb in victim.limbs)
+				victim.apply_damage(22, BRUTE, limb, sharp = TRUE)
+			for(var/obj/item/item in victim)
+				victim.drop_inv_item_to_loc(item, victim.loc, FALSE, TRUE)
+				victim.status_flags |= PERMANENTLY_DEAD
+			victim.add_flay_overlay(stage = 3)
+
+			//End the loop and remove all references to the datum.
+			current_flayer = null
+			UnregisterSignal(victim, COMSIG_HUMAN_FLAY_ATTEMPT)
+			victim = null
+			return
+
+	flay(target, user, tool)
```

Comment:
I'm confused as to why you flay here at the end

Reply:
i don't know but it work like that so...

Reply:
i try remove it and it gave me that message "DB is still starting up, please wait" when i try to go in edit character so i assume it's vital.

Reply:
so it gave me the same message even with it so i don't know what's happening there....

Reply:
if i remove it nothing work and i don't know where to move it....